File: lib/api/media.js
var _ = require('lodash'),
  Promise = require('bluebird'),
  util = require('../util'),
  log = util.logger();

function Media(hootsuite, connection) {
  this._hootsuite = hootsuite;
  this._connection = connection;
}

Media.prototype = {
  createUrl: function (sizeBytes, mimeType) {
    var path = util.createPath('media');
    var data = {
      sizeBytes: sizeBytes || null,
      mimeType: mimeType || 'video/mp4',
    };
    return this._connection.postJson(path, data);
  },
  
  statusById: function (mediaId) {
    var path = util.createPath('media', mediaId);
    return this._connection.get(path, { _method: 'GET' });
  }
}

module.exports = Media;

File: src.net/hootsuite-rest/Api/Media.cs
using Newtonsoft.Json.Linq;

namespace Hootsuite.Rest.Api
{
    public class Media
    {
        Hootsuite _hootsuite;
        Connection _connection;

        public Media(Hootsuite hootsuite, Connection connection)
        {
            _hootsuite = hootsuite;
            _connection = connection;
        }

        public JToken createUrl(int sizeBytes, string mimeType)
        {
            var path = util.createPath("media");
            var data = new
            {
                sizeBytes,
                mimeType = mimeType ?? "video/mp4",
            };
            return _connection.postJson(path, data);
        }

        public JToken statusById(string mediaId)
        {
            var path = util.createPath("media", mediaId);
            return _connection.get(path);
        }
    }
}
